tests/helper/helper_generic.go

```diff
@@ -177,6 +179,63 @@ func WatchNonRetCmdStdOut(cmdStr string, timeout time.Duration, check func(outpu
 	}
 }
 
+// RunCmdWithMatchOutputFromBuffer starts the command, and command stdout is attached to buffer.
+// we read data from buffer line by line, and if expected string is matched it returns true
+// It is different from WaitforCmdOut which gives stdout in one go using session.Out.Contents()
+// for commands like odo log -f which streams continuous data and does not terminate by their own
+// we need to read the stream data from buffer.
+func RunCmdWithMatchOutputFromBuffer(timeoutAfter time.Duration, matchString, program string, args ...string) (bool, error) {
```

Comment:
I was thinking instead of overloading helper file, Can't we use gingko in-build timeout for this - https://onsi.github.io/ginkgo/#asynchronous-tests

Reply:
@prietyc123 I follow the pattern from `odo watch` tests, timeout we are using this way only. I agree with your concern, once you update some other tests with ` gingko in-build` pattern as an example, i would follow the same.